### ISSUE TITLE
MapTiler Access Token does not get added to URLs

### ISSUE THREAD

**Issue by rsalzer** [maputnik/editor#869]
I tried to use a JSON style in which MapTiler-Datasource are used. I left all the key={key} in there in order for Maputnik to replace it when I add it in the "Style Setting".

Unfortunately it does not apply them.

Steps to reproduce:
* Open this file in Maputnik (Open --> Upload): 
[style.json](https://github.com/maplibre/maputnik/files/14101830/style.json)
* Add a MapTiler Access Token to the "Style Settings"

<img width="442" alt="image" src="https://github.com/maplibre/maputnik/assets/17248257/55edb92b-d274-4bf7-b8cb-b1e3301cda52">

**Comment by HarelM** [comment]
I would say that I hoped maptiler would be inclined to fix this, but they have their editor, so I'm not sure and help would come from there.
Having said that, maputnik codebase is small and you should be able to find and fix this issue.

**Comment by HarelM** [comment]
The code around replacing the access token is weird, and potentially buggy, or does not expect this flow.
Since the keys should be part of the style, I believe you should simply set them in the relevant source and not expect maputnik to replace it.

Bottom line, more of a feature than a bug :-)

**Comment by rsalzer** [comment]
Then the bottom line would be to remove those fields which suggest that it works ;)

A better flow would probably be to offer values which maputnik replaces ... {key} or {maptilerKey} and then replaces them everywhere.

**Comment by HarelM** [comment]
It works for glyphs and maybe sprites I believe.
But for sources there's a bunch of code that I didn't understand...

**Comment by rsalzer** [comment]
Probably it would be quite easy to add a possibility to add variables (in the shape of {variable}) which then can get replaced. Like this it would be universal ... I might add do a PR

**Comment by birkskyum** [comment]
Why don't we just keep the api keys in the urls instead of all this find/replace {key} complexity?

**Comment by HarelM** [comment]
Yes, the glyphs, sprite and source URLs can be added as a full address including the access token and avoid all this complexity...

**Comment by birkskyum** [comment]
@HarelM , I realized it's maybe not great to only have the url.. if you have a style locally with urls containing {key} and upload that, it's necessary that Maputnik can replace it, rather than expect some server to do it.

**Comment by HarelM** [comment]
Ahh, I see... I don't have a good idea how to solve this from the top of my head, it might require more thought to think about a proper UX to solve this on one hand and easier maintenance on the other...
Or that simply there's a need to handle this differently in the code, maybe using transfrom url or something, IDK...

**Comment by bbmaps24** [comment]
can someone please highlight where in the code I need to adjust and add the mapbox token? mapbox is telling me the source and URL are undefined. I suspect the issue is somewhere in this part of the code (the rest is all styles but I did leave park style in there in case that part needs to be adjusted too and I can replicate it elsewhere)

{
  "version": 8,
  "name": "Test",
  "metadata": {},
  "sources": {
    "ne2_shaded": {
      "maxzoom": 6,
      "tileSize": 256,
      "tiles": [
        "https://tiles.openfreemap.org/natural_earth/ne2sr/{z}/{x}/{y}.png"
      ],
      "type": "raster"
    },
    "openmaptiles": {
      "type": "vector",
      "url": "https://tiles.openfreemap.org/planet"
    }
  },
  "sprite": "https://tiles.openfreemap.org/sprites/ofm_f384/ofm",
  "glyphs": "https://tiles.openfreemap.org/fonts/{fontstack}/{range}.pbf",
  "layers": [
    {
      "id": "background",
      "type": "background",
      "layout": {"visibility": "none"},
      "paint": {"background-color": "rgb(242,243,240)"}
    },
    {
      "id": "park",
      "type": "fill",
      "source": "openmaptiles",
      "source-layer": "park",
      "filter": [
        "match",
        ["geometry-type"],
        ["MultiPolygon", "Polygon"],
        true,
        false
      ],
      "layout": {"visibility": "none"},
      "paint": {"fill-color": "rgb(230, 233, 229)"}
    },

**Comment by HarelM** [comment]
openfreemap doesn't need a token...

**Comment by bbmaps24** [comment]
when I try to upload the json style to mapbox I get this error

<img width="2160" height="1352" alt="Image" src="https://github.com/user-attachments/assets/9a3c9879-e85d-4731-94af-02ddd9776a19" />

**Comment by HarelM** [comment]
Please contact mapbox support then...
I would recommend starting from a style in maputnik and adjust it as needed, or read more about the echo system...

**Comment by bbmaps24** [comment]
ok so it is not related to tokens/the code export from maputnik? 

this was a standard maputnik style that I edited.

**Comment by HarelM** [comment]
No.
